### PR TITLE
GraphRoles: More concrete details

### PR DESCRIPTION
**Pull request by glasser**

### CODE REVIEW

File: preview-docs/GraphRoles.md [hunk 1]
```diff
@@ -8,11 +8,11 @@ We've identified new requirements for access control at the graph level since th
...
-4. Introduce a way to assign roles to graph API keys. We would make the **Consumer**, **Observer**, **Contributor**, and **Graph Admin** roles available for assignment on API key creation.
-5. Introduce a setting on graphs that flags them as "visible" or "hidden" to the members of that graph's org by default. Right now, all graphs in the org are visible to all members of the org. If you mark a graph as "hidden" to your org, only you will see it. You can grant permission for other users in your org to see your graph as well.
+1. Introduce a **Graph Admin** role that can operate with all permissions of the **Contributor** role, but can also manage user access to the graph and delete the graph. (Our current **Admin** role will be renamed to **Org Admin**. **Graph Admin**s and **Org Admin**s will have identical permissions to act on graphs, but org admins will also be able to manage org membership, billing, etc.)  The **Contributor** role will be made less powerful. It will not be able to perform management tasks on the graph like updating integrations and settings; it will have all the permissions of **Observer** plus the ability to push new versions of the graph's schema.
```
Comment (ndintenfass) [inline preview-docs/GraphRoles.md:11]: Do we have a sense of the migration for existing Contributors? Do they become graph Admins or lose permissions?

File: preview-docs/GraphRoles.md [hunk 2]
```diff
@@ -8,11 +8,11 @@ We've identified new requirements for access control at the graph level since th
...
+2. Introduce a way to override any role a user might have in the org with a higher role for a specific graph. A user might be an **Observer** in the org, but can be promoted to a **Contributor** on a specific graph. When users create a graph, they are automatically made **Graph Admin** on that graph.
+3. Introduce a way to mark variants as **protected**. On a protected variant, you need to be a **Graph Admin** rather than a **Contributor** to push new versions of the graph's schema.
```
Comment (ndintenfass) [inline preview-docs/GraphRoles.md:13]: For extra clarity, consider adding "to that variant" at the end of this sentence.

**Review by ndintenfass — APPROVED** [review]
Left a couple suggestions inline.

File: preview-docs/GraphRoles.md (hunk 1 above)
Comment (glasser) [reply preview-docs/GraphRoles.md:11]: I think it's worth reaching out to active customers to see what they'd prefer, but yeah, probably transitioning to Graph Admin (which will be identical to the old Contributor except that they will be able to delete graphs and manage per-graph permissions) makes sense. What do you think?

**Review by daniman — APPROVED** [review]

**Review by Jephuff — APPROVED** [review]

File: preview-docs/GraphRoles.md (hunk 1 above)
Comment (ndintenfass) [reply preview-docs/GraphRoles.md]: Sorry for the late reply - ideally, I think we'd pull which customers have Contributors and let them choose what to do -- keep them as less-permissioned Contributors or upgrade them to Graph Admins. In general, if we have to take the expedient route it's probably better to keep them as Contributors, take away those permissions, then tell them all they can up-level those users at their discretion. Better to take away permissions than to give them. cc @daniman for discussion on implementing the comms on this - might ask you to run with this, or at least help me get the list.

**Pull request merged**